src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/ComponentBroadcaster.cs

```diff
@@ -182,56 +182,56 @@ private void EnsureComponentInitialized()
             }
         }
 
-        protected virtual void SendComponentCreation(IEnumerable<SocketEndpoint> newConnections)
+        protected virtual void SendComponentCreation(IEnumerable<INetworkConnection> newConnections)
         {
-            if (endpointsNeedingComponentCreation != null)
+            if (connectionsNeedingComponentCreation != null)
             {
-                for (int i = endpointsNeedingComponentCreation.Count - 1; i >= 0; i--)
+                for (int i = connectionsNeedingComponentCreation.Count - 1; i >= 0; i--)
                 {
-                    if (ShouldSendChanges(endpointsNeedingComponentCreation[i]))
+                    if (ShouldSendChanges(connectionsNeedingComponentCreation[i]))
                     {
-                        SendComponentCreation(endpointsNeedingComponentCreation[i]);
-                        endpointsNeedingComponentCreation.RemoveAt(i);
+                        SendComponentCreation(connectionsNeedingComponentCreation[i]);
+                        connectionsNeedingComponentCreation.RemoveAt(i);
                     }
                 }
             }
 
-            foreach (SocketEndpoint newConnection in newConnections)
+            foreach (INetworkConnection newConnection in newConnections)
             {
                 if (ShouldSendChanges(newConnection))
                 {
                     SendComponentCreation(newConnection);
                 }
                 else
                 {
-                    if (endpointsNeedingComponentCreation == null)
+                    if (connectionsNeedingComponentCreation == null)
                     {
-                        endpointsNeedingComponentCreation = new List<SocketEndpoint>();
+                        connectionsNeedingComponentCreation = new List<INetworkConnection>();
                     }
-                    endpointsNeedingComponentCreation.Add(newConnection);
+                    connectionsNeedingComponentCreation.Add(newConnection);

```

Comment:
```
(connectionsNeedingComponentCreation ??= new List<INetworkConnection>())
    .Add(newConnection);
```

Reply:
I wasn't sure if ??= was an encoding error or a misplaced C++ trigraph! Turns out it's C# 8.0. Would there be any gotchas with bumping the langversion?  https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

Reply:
If it's not already supported by Unity, then that's probably outside of our control.

Reply:
keeping this this way for now. This would be a good suggestion in the future.